cmd/zoekt-indexserver/main.go

```diff
@@ -141,7 +141,7 @@ func fetchGitRepo(dir string) bool {
 		log.Printf("command %s failed: %v\nOUT: %s\nERR: %s",
 			cmd.Args, err, outBuf.String(), errBuf.String())
 	} else {
-		return len(outBuf.Bytes()) != 0
+		return len(errBuf.Bytes()) != 0
 	}
 	return false
```

Comment:
The code (not your fix) is a bit janky. Would you mind cleaning it up a bit while your are at it? EG we could remove the `else` block and return early if `cmd.Run()` returns with an error.

Reply:
Don't mind at all! I'll give it a shot later today.

Reply:
Alright, went a little further than just changing the else block, let me know what you think.

Here's a screenshot from some print debugs to make sure it works as expected. 

![Screenshot 2023-06-27 at 10 42 22 AM](https://github.com/sourcegraph/zoekt/assets/23644345/9c24cefc-e483-4a23-8a37-68f9694a7057)

One thing to note for the future, as far as I can tell `fetch` prints errors to stderr as well sometimes, if we really want to be fancy we can probably switch to the `--porcelain` ([introduced here](https://github.com/git/git/commit/dd781e3856bccd3793122f7f4e604e5a89ae517d)) mode to ensure that what's printed is actually an update...

Reply:
> One thing to note for the future, as far as I can tell fetch prints actual errors to stderr as well sometimes, if we really want to be fancy we can probably switch to the --porcelain ([introduced here](https://github.com/git/git/commit/dd781e3856bccd3793122f7f4e604e5a89ae517d)) mode to ensure that what's printed is actually an update...

That looks nice indeed!